BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Tracing/SelfDiagnostics/SelfDiagnosticsConfigParser.cs

```diff
@@ -31,11 +36,149 @@ internal class SelfDiagnosticsConfigParser
         // In theory the variable won't be access at the same time because worker thread first Task.Delay for a few seconds.
         private byte[] configBuffer;
 
+        /// <summary>
+        /// Represents the location for App Insights to parse user-defined self-diagnostics settings.
+        /// </summary>
+        internal enum ParseLocation
+        {
+            /// <summary>
+            /// Parse self-diagnostics settings from enviornment variable(s).
+            /// </summary>
+            EnviornmentVariable,
+
+            /// <summary>
+            /// Parse self-diagnostics settings from tje JSON file.
+            /// </summary>
+            ConfigJson,
+        }
+
         public bool TryGetConfiguration(out string logDirectory, out int fileSizeInKB, out EventLevel logLevel)
+        {
+            if (TryGetConfigFromEnvrionmentVariable(out logDirectory, out fileSizeInKB, out logLevel))
+            {
+                // self-diagnostics settings passed in via enviornment variables has higher precedence than
+                // self-diagnostcis settings passed in via JSON file.
+                return true;
+            }
+
+            return this.TryGetConfigFromJsonFile(ref logDirectory, ref fileSizeInKB, ref logLevel);
+        }
+
+        internal static bool TryGetConfigFromEnvrionmentVariable(out string logDirectory, out int fileSizeInKB, out EventLevel logLevel)
         {
             logDirectory = null;
             fileSizeInKB = 0;
             logLevel = EventLevel.LogAlways;
```

Comment:
These are not necessary, as you are going to reset each value in each `TryParse*` method.

Reply:
Yeah, I agree that the design of parsing/try-catching logic could be better but now sure how yet.

For the current design, if we don't assign the out parameters a default value, when control reaches line 75, compiler would complain that the out parameters must be assigned before control leaves the current method.